core/embed/rust/src/ui/model_mercury/flow/confirm_output.rs

```diff
@@ -69,68 +113,302 @@ impl FlowState for ConfirmOutput {
     }
 }
 
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum ConfirmOutputWithSummary {
+    Address,
+    AddressMenu,
+    AddressMenuCancel,
+    Summary,
+    SummaryMenu,
+    SummaryMenuCancel,
+    FeeInfo,
+    Hold,
+    HoldMenu,
+    HoldMenuCancel,
+    AccountInfo,
+}
+
+impl FlowState for ConfirmOutputWithSummary {
+    #[inline]
+    fn index(&'static self) -> usize {
+        *self as usize
+    }
+
+    fn handle_swipe(&'static self, direction: SwipeDirection) -> StateChange {
+        match (self, direction) {
+            (Self::Address, SwipeDirection::Left) => Self::AddressMenu.swipe(direction),
+            (Self::Address, SwipeDirection::Up) => Self::Summary.swipe(direction),
+            (Self::AccountInfo, SwipeDirection::Right) => Self::AddressMenu.swipe(direction),
+            (Self::Summary, SwipeDirection::Left) => Self::SummaryMenu.swipe(direction),
+            (Self::Summary, SwipeDirection::Up) => Self::Hold.swipe(direction),
+            (Self::Summary, SwipeDirection::Down) => Self::Address.swipe(direction),
+            (Self::FeeInfo, SwipeDirection::Right) => Self::SummaryMenu.swipe(direction),
+            (Self::Hold, SwipeDirection::Left) => Self::HoldMenu.swipe(direction),
+            (Self::Hold, SwipeDirection::Down) => Self::Summary.swipe(direction),
+            _ => self.do_nothing(),
+        }
+    }
+
+    fn handle_event(&'static self, msg: FlowMsg) -> StateChange {
+        match (self, msg) {
+            (Self::Address, FlowMsg::Info) => Self::AddressMenu.transit(),
+            (Self::AddressMenu, FlowMsg::Choice(MENU_ITEM_CANCEL)) => {
+                Self::AddressMenuCancel.swipe_left()
+            }
+            (Self::AddressMenuCancel, FlowMsg::Cancelled) => Self::AddressMenu.swipe_right(),
+            (Self::Summary, FlowMsg::Info) => Self::SummaryMenu.transit(),
+            (Self::SummaryMenu, FlowMsg::Choice(MENU_ITEM_CANCEL)) => {
+                Self::SummaryMenuCancel.swipe_left()
+            }
+            (Self::SummaryMenuCancel, FlowMsg::Cancelled) => Self::SummaryMenu.swipe_right(),
+            (Self::Hold, FlowMsg::Info) => Self::HoldMenu.transit(),
+            (Self::HoldMenu, FlowMsg::Choice(MENU_ITEM_CANCEL)) => {
+                Self::HoldMenuCancel.swipe_left()
+            }
+            (Self::HoldMenuCancel, FlowMsg::Cancelled) => Self::HoldMenu.swipe_right(),
+            (
+                Self::AddressMenu | Self::SummaryMenu | Self::HoldMenu,
```

Comment:
Probably unnecessary to specify all `Menu` if the option is only present in `Summary`, no?

Same for `AccountInfo` below.

Reply:
https://github.com/trezor/trezor-firmware/pull/4095/commits/ec6a2a8c8d7e840f3b19e2ab09c9bb08806c1449#diff-5cb317ed79ea16b58238d2743af510c9f82a27c2764811d83f532f8d8358b6f6R169